lib/backend/etcd/etcd.go

```diff
@@ -120,6 +121,35 @@ func (c *EtcdClient) EnsureInitialized() error {
 		log.Info("Ready flag is already set")
 	}
 
+	// The confd agent in calico/node requires a few directories to exist
+	// to be able to monitor correctly:
+	// -  IP pools parent directories
+	// -  BGP custom filters directories.
+	if err := c.ensureDirectory("/calico/v1/ipam/v4/pool"); err != nil {
```

Comment:
Exactly!  This is only because confd watches directories :-(

Ideally we only really want to do this configuration when using confd.  One thing I'd considered was to have a "sneaky" back door "backend" API that we hook into just for confd - in fact most of this process is only really required when using calico/node and confd.  it's a little naughty, but would be a temporary measure until we expand our API sufficiently to be able to drop confd.   WDYT?